docs/fspiop-rest-v1.0-OpenAPI-implementation_openapi3.yaml

```diff
@@ -0,0 +1,3965 @@
+      - $ref: '#/components/parameters/X-Forwarded-For'
+      - $ref: '#/components/parameters/FSPIOP-Source'
+      - $ref: '#/components/parameters/FSPIOP-Destination'
+      - $ref: '#/components/parameters/FSPIOP-Encryption'
+      - $ref: '#/components/parameters/FSPIOP-Signature'
+      - $ref: '#/components/parameters/FSPIOP-URI'
+      - $ref: '#/components/parameters/FSPIOP-HTTP-Method'
+      requestBody:
+        description: Details of the error returned.
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/ErrorInformationObject'
+      responses:
+        200:
+          $ref: '#/components/responses/200'
+        400:
+          $ref: '#/components/responses/400'
+        401:
+          $ref: '#/components/responses/401'
+        403:
+          $ref: '#/components/responses/403'
+        404:
+          $ref: '#/components/responses/404'
+        405:
+          $ref: '#/components/responses/405'
+        406:
+          $ref: '#/components/responses/406'
+        501:
+          $ref: '#/components/responses/501'
+        503:
+          $ref: '#/components/responses/503'
+
+components:
+  schemas:
+  #Element definitions
+    Amount:
+      title: Amount
+      type: string
+      pattern: ^([0]|([1-9][0-9]{0,17}))([.][0-9]{0,3}[1-9])?$
+      description: The API data type Amount is a JSON String in a canonical format that is restricted by a regular expression for interoperability reasons. This pattern does not allow any trailing zeroes at all, but allows an amount without a minor currency unit. It also only allows four digits in the minor currency unit; a negative value is not allowed. Using more than 18 digits in the major currency unit is not allowed.
+    AmountType:
+      title: AmountType
+      type: string
+      enum:
+      - SEND
+      - RECEIVE
+      description: Below are the allowed values for the enumeration AmountType.
+      
+        - SEND - Amount the Payer would like to send, that is, the amount that should be withdrawn from the Payer account including any fees.
+        
+        - RECEIVE - Amount the Payer would like the Payee to receive, that is, the amount that should be sent to the receiver exclusive of any fees.    
+    AuthenticationType:
+      title: AuthenticationType
+      type: string
+      enum:
+      - OTP
+      - QRCODE
+      - U2F
+      description: Below are the allowed values for the enumeration AuthenticationType.
+      
+        - OTP - One-time password generated by the Payer FSP.
+        
+        - QRCODE - QR code used as One Time Password.
+
+        - U2F challenge-response, where payer FSP verifies if the response provided by end-user device matches the previously registered key.
+    
+    AuthenticationValue:
+      title: AuthenticationValue
+      oneOf:
+        - $ref: '#/components/schemas/OtpValue'
+        - $ref: '#/components/schemas/QRCODE'

```

Comment:
Should this be `QRCodeValue` to match the others?

Reply:
That is existing definition , if you wish I will change it.

Reply:
Sorry I thought you were adding a new one. Just leave it then.